### ISSUE TITLE
Could more time be given before removal?

### ISSUE THREAD

**Issue by sanbox-irl** [bitshifter/glam-rs#111]
Hi there -- I hope this isn't too rude, but would it be possible to have more time between deprecating and removing functions? The recent changes to Vec2s and 3s effectively went from stable to removed in one week, which is relatively fast.

Mind you, it was such a simple change that it wasn't hard to fix across a reasonably large game project, but perhaps it could have received a full major path version in deprecation before removal?

I am completely aware that I'm asking you to go a bit above and beyond the call of semver, since we're pre-1.0.0, and it also occurs to me that this deprecation is probably a result of you simply doing so much great work on this great project, so you're really just getting updates out quick, which I appreciate a lot.

Thanks so much!

**Comment by bitshifter** [comment]
Which changes are you talking about? Do you mean the removal of accessors like `.x()`, `.set_x()`, and `.x_mut()` etc.?

I did make a number of releases that if you pinned to the right version compilation should have worked. Version 0.10.1 had both accessors  and the new direct access, 0.10.2 deprecated the accessors and 0.11.0 removed the methods. Generally I use the semver MINOR version number for breaking changes and I try to deprecated in a previous MINOR version prior to making the breaking changes. The semver PATCH version should be for non breaking changes.

If you pin to a particular MINOR version then I would hope you would not get any breaking changes.

All of these updates were in https://github.com/bitshifter/glam-rs/blob/master/CHANGELOG.md.

**Comment by bitshifter** [comment]
Closing this as it's more of a discussion, but I would be interested to know why staging the releases as I described above didn't work for you.

**Comment by sanbox-irl** [comment]
I actually think staging it was totally fine, but in realtime, that happened very fast actually. It was only a week! I actually was pinned to a minor version, saw there was a new version, and then updated (skipping the deprecation time entirely) and got the errors from the removal of the accessors. 

Although I agree that you followed a good procedure, because it happened on such a compressed time frame, a week isn't much time. It quite trivial to miss such a small deprecation period.

I think a reasonable argument to this would be just "don't update then!", but still, it seems off to me a bit, even though you followed a good procedure

**Comment by bitshifter** [comment]
In this particular case I saw some people complaining about compiler errors in the versions where both the accessors methods and direct access were co-existing. I didn't get these errors myself but I could see how they might happen. That's the main reason I removed them quite quickly. Usually I leave deprecated methods around for a while. Until glam is 1.0 though, I will try not to break things on people but I make no guarantees.
